fly/integration/checklist_test.go

```diff
@@ -141,6 +141,60 @@ some-orphaned-job: concourse.check %s main some-pipeline/branch:master some-orph
 job-1: concourse.check %s main some-pipeline/branch:master job-1
 job-2: concourse.check %s main some-pipeline/branch:master job-2
 
+`, atcServer.URL(), atcServer.URL())))
+				})
+			})
+		})
+
+		Context("when a team name is specified alongwith pipeline name", func() {
+			JustBeforeEach(func() {
+				atcServer.AppendHandlers(
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/api/v1/teams/other-team"),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, atc.Team{
+							Name: "other-team",
+						}),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/api/v1/teams/other-team/pipelines/some-pipeline/config"),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, atc.ConfigResponse{Config: config}, http.Header{atc.ConfigVersionHeader: {"42"}}),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("PUT", "/api/v1/teams/other-team/pipelines/some-pipeline/config"),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, atc.ConfigResponse{Config: config}, http.Header{atc.ConfigVersionHeader: {"42"}}),
+					),
+				)
+			})
+
+			Context("when there are no groups", func() {
+				BeforeEach(func() {
+					config = atc.Config{
+						Jobs: atc.JobConfigs{
+							{
+								Name: "job-1",
+							},
+							{
+								Name: "job-2",
+							},
+						},
+					}
+
+				})
```

Comment:
I think could be a good idea to get ride of this context here because:
1) It is not clear what the test is doing at this point
2) Whether groups are part of the response has nothing to do with the team flag being used

You could just delete that context and just move the IT section outside of the context, at the same level of the JustBeforeEach function.

You could also move the `config` inside the BeforeEach to be part of the JustBeforeEach so the test can still pass.

Reply:
Thanks for the review! I have updated the test!